pkg/github/context_tools.go

```diff
@@ -2,52 +2,39 @@ package github
 
 import (
 	"context"
-	"encoding/json"
-	"fmt"
-	"io"
-	"net/http"
 
 	"github.com/github/github-mcp-server/pkg/translations"
 	"github.com/mark3labs/mcp-go/mcp"
 	"github.com/mark3labs/mcp-go/server"
 )
 
 // GetMe creates a tool to get details of the authenticated user.
-func GetMe(getClient GetClientFn, t translations.TranslationHelperFunc) (tool mcp.Tool, handler server.ToolHandlerFunc) {
-	return mcp.NewTool("get_me",
-			mcp.WithDescription(t("TOOL_GET_ME_DESCRIPTION", "Get details of the authenticated GitHub user. Use this when a request includes \"me\", \"my\". The output will not change unless the user changes their profile, so only call this once.")),
-			mcp.WithToolAnnotation(mcp.ToolAnnotation{
-				Title:        t("TOOL_GET_ME_USER_TITLE", "Get my user profile"),
-				ReadOnlyHint: toBoolPtr(true),
-			}),
-			mcp.WithString("reason",
-				mcp.Description("Optional: the reason for requesting the user information"),
-			),
+func GetMe(getClient GetClientFn, t translations.TranslationHelperFunc) (mcp.Tool, server.ToolHandlerFunc) {
+	tool := mcp.NewTool("get_me",
+		mcp.WithDescription(t("TOOL_GET_ME_DESCRIPTION", "Get details of the authenticated GitHub user. Use this when a request includes \"me\", \"my\". The output will not change unless the user changes their profile, so only call this once.")),
+		mcp.WithToolAnnotation(mcp.ToolAnnotation{
+			Title:        t("TOOL_GET_ME_USER_TITLE", "Get my user profile"),
+			ReadOnlyHint: toBoolPtr(true),
+		}),
+		mcp.WithString("reason",
+			mcp.Description("Optional: the reason for requesting the user information"),
 		),
-		func(ctx context.Context, _ mcp.CallToolRequest) (*mcp.CallToolResult, error) {
-			client, err := getClient(ctx)
-			if err != nil {
-				return nil, fmt.Errorf("failed to get GitHub client: %w", err)
-			}
-			user, resp, err := client.Users.Get(ctx, "")
-			if err != nil {
-				return nil, fmt.Errorf("failed to get user: %w", err)
-			}
-			defer func() { _ = resp.Body.Close() }()
+	)
 
-			if resp.StatusCode != http.StatusOK {
-				body, err := io.ReadAll(resp.Body)
-				if err != nil {
-					return nil, fmt.Errorf("failed to read response body: %w", err)
-				}
-				return mcp.NewToolResultError(fmt.Sprintf("failed to get user: %s", string(body))), nil
-			}
-
-			r, err := json.Marshal(user)
-			if err != nil {
-				return nil, fmt.Errorf("failed to marshal user: %w", err)
-			}
+	type args struct{}
+	handler := mcp.NewTypedToolHandler(func(ctx context.Context, _ mcp.CallToolRequest, _ args) (*mcp.CallToolResult, error) {
+		client, err := getClient(ctx)
```

Comment:
I have some ideas about having a handler function that injects:

```go
type Clients struct {
 rest..
 gql..
}
```

Then all handlers don't have this `getClient` boilerplate. Added a lot of types though, would rather feel a bit more pain.